{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "9533c514",
   "metadata": {},
   "source": [
    "# –ü—Ä–æ–µ–∫—Ç –ø–æ –∏—Ç–æ–≥–∞–º –∫—É—Ä—Å–∞"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c8653801",
   "metadata": {},
   "source": [
    "–°–∫–∞—á–∞–π—Ç–µ –¥–∞–Ω–Ω—ã–µ SRR12905311 –∏–∑ SRA NCBI. –û–ø–∏—à–∏—Ç–µ, —á—Ç–æ –∑–∞ –∏—Å—Å–ª–µ–¥–æ–≤–∞–Ω–∏–µ –ø—Ä–æ–≤–æ–¥–∏–ª–æ—Å—å, –Ω–∞ –∫–∞–∫–æ–º –æ—Ä–≥–∞–Ω–∏–∑–º–µ. –ú–æ–∂–Ω–æ –≤–æ—Å–ø–æ–ª—å–∑–æ–≤–∞—Ç—å—Å—è GEO (GSM4862160).  \n",
    "–ü—Ä–µ–ø—Ä–æ—Ü–µ—Å—Å–∏—Ä—É–π—Ç–µ —Ä–∏–¥—ã (–æ—Ü–µ–Ω–∫–∞ –∫–∞—á–µ—Å—Ç–≤–∞, —Ç—Ä–∏–º–º–∏—Ä–æ–≤–∞–Ω–∏–µ –ø–æ –Ω–µ–æ–±—Ö–æ–¥–∏–º–æ—Å—Ç–∏, –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏–µ –Ω–∞ —Ä–µ—Ñ–µ—Ä–µ–Ω—Å), –∏ –ø—Ä–æ–≤–µ–¥–∏—Ç–µ –ø–æ–∏—Å–∫ SNP.  \n",
    "–í—ã–±–µ—Ä–∏—Ç–µ 3-4 –≤—ã—è–≤–ª–µ–Ω–Ω—ã–µ –º—É—Ç–∞—Ü–∏–∏. –ü—Ä–æ–≤–µ—Ä—å—Ç–µ –ø–æ NCBI, –∫–∞–∫–∏–º –≥–µ–Ω–∞–º –æ–Ω–∏ –ø—Ä–∏–Ω–∞–¥–ª–µ–∂–∞—Ç, –∏ –ø—Ä–µ–¥–ø–æ–ª–æ–∂–∏—Ç–µ, –∫–∞–∫ —ç—Ç–∏ –º—É—Ç–∞—Ü–∏–∏ –º–æ–≥—É—Ç –≤–ª–∏—è—Ç—å –Ω–∞ –∂–∏–∑–Ω–µ–¥–µ—è—Ç–µ–ª—å–Ω–æ—Å—Ç—å –æ—Ä–≥–∞–Ω–∏–∑–º–∞.  \n",
    "–ù–µ –∑–∞–±—ã–≤–∞–π—Ç–µ, —á—Ç–æ –≤–∞—à–∏ –º—É—Ç–∞—Ü–∏–∏ –Ω–µ –æ—Ç–Ω–æ—Å—è—Ç—Å—è –∫ —á–µ–ª–æ–≤–µ–∫—É."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "62e60ea9",
   "metadata": {},
   "outputs": [],
   "source": [
    "# !sudo apt install fastp fastqc multiqc\n",
    "# !sudo apt install spades; pip3 install quast\n",
    "# !sudo apt install samtools bwa\n",
    "# !sudo apt install kraken2 centrifuge\n",
    "# !sudo apt install samtools bamtools freebayes bedtools tabix bcftools  # libvcflib-tools\n",
    "# #!sudo apt install busco augustus prokka igv\n",
    "# !sudo apt install mafft raxml iqtree bedtools\n",
    "# !sudo apt install snpeff genometools bedtools\n",
    "# #!sudo apt install ea-utils sickle seqtk bowtie2 muscle"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "1ffeccbc",
   "metadata": {
    "scrolled": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "+ ORGANISM=PAeruginosa\n",
      "+ GENOME=GCF_000006765.1\n",
      "+ RUN=SRR12905311\n",
      "+ ABRICATE=/home/nikolay/Apps/abricate/bin/abricate\n",
      "+ QUALIMAP=/home/nikolay/Apps/qualimap_v2.3/qualimap\n",
      "+ RESULT_DIR=/mnt/c/Users/Asus/BioGen\n",
      "+ '[' -d /mnt/c/Users/Asus/BioGen ']'\n",
      "+ mkdir -p reads ref\n",
      "+ '[' -f reads/SRR12905311_1.fastq.gz ']'\n",
      "+ '[' -f reads/SRR12905311_2.fastq.gz ']'\n",
      "+ '[' -f GCF_000006765.1.zip ']'\n",
      "+ unzip GCF_000006765.1.zip\n",
      "Archive:  GCF_000006765.1.zip\n",
      "  inflating: README.md               \n",
      "  inflating: ncbi_dataset/data/assembly_data_report.jsonl  \n",
      "  inflating: ncbi_dataset/data/GCF_000006765.1/GCF_000006765.1_ASM676v1_genomic.fna  \n",
      "  inflating: ncbi_dataset/data/GCF_000006765.1/genomic.gff  \n",
      "  inflating: ncbi_dataset/data/GCF_000006765.1/cds_from_genomic.fna  \n",
      "  inflating: ncbi_dataset/data/GCF_000006765.1/protein.faa  \n",
      "  inflating: ncbi_dataset/data/GCF_000006765.1/sequence_report.jsonl  \n",
      "  inflating: ncbi_dataset/data/dataset_catalog.json  \n",
      "+ mv ncbi_dataset/data/GCF_000006765.1/GCF_000006765.1_ASM676v1_genomic.fna ref/PAeruginosa.fasta\n",
      "+ mv ncbi_dataset/data/GCF_000006765.1/genomic.gff ref/PAeruginosa.gff\n",
      "+ rm -rf ncbi_dataset README.md\n",
      "+ rm -rf quality_check\n",
      "+ mkdir -p quality_check\n",
      "+ mkdir -p trimmed\n",
      "++ nproc\n",
      "+ fastp --overrepresentation_analysis --trim_poly_g -f 12 -t 2 -h quality_check/SRR12905311.fastp.html -j quality_check/SRR12905311.fastp.json --thread 12 -i reads/SRR12905311_1.fastq.gz -I reads/SRR12905311_2.fastq.gz -o trimmed/SRR12905311_1.fastq.gz -O trimmed/SRR12905311_2.fastq.gz\n",
      "Read1 before filtering:\n",
      "total reads: 6371441\n",
      "total bases: 506337238\n",
      "Q20 bases: 489048202(96.5855%)\n",
      "Q30 bases: 481766970(95.1474%)\n",
      "\n",
      "Read2 before filtering:\n",
      "total reads: 6371441\n",
      "total bases: 506091161\n",
      "Q20 bases: 483530535(95.5422%)\n",
      "Q30 bases: 476834184(94.219%)\n",
      "\n",
      "Read1 after filtering:\n",
      "total reads: 6178857\n",
      "total bases: 404396781\n",
      "Q20 bases: 391989721(96.932%)\n",
      "Q30 bases: 386168635(95.4925%)\n",
      "\n",
      "Read2 aftering filtering:\n",
      "total reads: 6178857\n",
      "total bases: 404415205\n",
      "Q20 bases: 391029873(96.6902%)\n",
      "Q30 bases: 385989618(95.4439%)\n",
      "\n",
      "Filtering result:\n",
      "reads passed filter: 12357714\n",
      "reads failed due to low quality: 345680\n",
      "reads failed due to too many N: 5118\n",
      "reads failed due to too short: 34370\n",
      "reads with adapter trimmed: 35136\n",
      "bases trimmed due to adapters: 198873\n",
      "\n",
      "Duplication rate: 2.43908%\n",
      "\n",
      "Insert size peak (evaluated by paired-end reads): 24\n",
      "\n",
      "JSON report: quality_check/SRR12905311.fastp.json\n",
      "HTML report: quality_check/SRR12905311.fastp.html\n",
      "\n",
      "fastp --overrepresentation_analysis --trim_poly_g -f 12 -t 2 -h quality_check/SRR12905311.fastp.html -j quality_check/SRR12905311.fastp.json --thread 12 -i reads/SRR12905311_1.fastq.gz -I reads/SRR12905311_2.fastq.gz -o trimmed/SRR12905311_1.fastq.gz -O trimmed/SRR12905311_2.fastq.gz \n",
      "fastp v0.20.1, time used: 26 seconds\n",
      "+ firefox quality_check/SRR12905311.fastp.html\n",
      "+ FASTQ_DIR=trimmed\n",
      "+ fastqc -o quality_check trimmed/SRR12905311_1.fastq.gz trimmed/SRR12905311_2.fastq.gz ../SRR12905311.fastq.gz\n",
      "Skipping '../SRR12905311.fastq.gz' which didn't exist, or couldn't be read\n",
      "Started analysis of SRR12905311_1.fastq.gz\n",
      "Approx 5% complete for SRR12905311_1.fastq.gz\n",
      "Approx 10% complete for SRR12905311_1.fastq.gz\n",
      "Approx 15% complete for SRR12905311_1.fastq.gz\n",
      "Approx 20% complete for SRR12905311_1.fastq.gz\n",
      "Approx 25% complete for SRR12905311_1.fastq.gz\n",
      "Approx 30% complete for SRR12905311_1.fastq.gz\n",
      "Approx 35% complete for SRR12905311_1.fastq.gz\n",
      "Approx 40% complete for SRR12905311_1.fastq.gz\n",
      "Approx 45% complete for SRR12905311_1.fastq.gz\n",
      "Approx 50% complete for SRR12905311_1.fastq.gz\n",
      "Approx 55% complete for SRR12905311_1.fastq.gz\n",
      "Approx 60% complete for SRR12905311_1.fastq.gz\n",
      "Approx 65% complete for SRR12905311_1.fastq.gz\n",
      "Approx 70% complete for SRR12905311_1.fastq.gz\n",
      "Approx 75% complete for SRR12905311_1.fastq.gz\n",
      "Approx 80% complete for SRR12905311_1.fastq.gz\n",
      "Approx 85% complete for SRR12905311_1.fastq.gz\n",
      "Approx 90% complete for SRR12905311_1.fastq.gz\n",
      "Approx 95% complete for SRR12905311_1.fastq.gz\n",
      "Analysis complete for SRR12905311_1.fastq.gz\n",
      "Started analysis of SRR12905311_2.fastq.gz\n",
      "Approx 5% complete for SRR12905311_2.fastq.gz\n",
      "Approx 10% complete for SRR12905311_2.fastq.gz\n",
      "Approx 15% complete for SRR12905311_2.fastq.gz\n",
      "Approx 20% complete for SRR12905311_2.fastq.gz\n",
      "Approx 25% complete for SRR12905311_2.fastq.gz\n",
      "Approx 30% complete for SRR12905311_2.fastq.gz\n",
      "Approx 35% complete for SRR12905311_2.fastq.gz\n",
      "Approx 40% complete for SRR12905311_2.fastq.gz\n",
      "Approx 45% complete for SRR12905311_2.fastq.gz\n",
      "Approx 50% complete for SRR12905311_2.fastq.gz\n",
      "Approx 55% complete for SRR12905311_2.fastq.gz\n",
      "Approx 60% complete for SRR12905311_2.fastq.gz\n",
      "Approx 65% complete for SRR12905311_2.fastq.gz\n",
      "Approx 70% complete for SRR12905311_2.fastq.gz\n",
      "Approx 75% complete for SRR12905311_2.fastq.gz\n",
      "Approx 80% complete for SRR12905311_2.fastq.gz\n",
      "Approx 85% complete for SRR12905311_2.fastq.gz\n",
      "Approx 90% complete for SRR12905311_2.fastq.gz\n",
      "Approx 95% complete for SRR12905311_2.fastq.gz\n",
      "Analysis complete for SRR12905311_2.fastq.gz\n",
      "+ firefox quality_check/SRR12905311_1_fastqc.html quality_check/SRR12905311_2_fastqc.html\n",
      "+ rm -rf multiqc_data multiqc_report.html\n",
      "+ multiqc quality_check trimmed -o quality_check\n",
      "\n",
      "  \u001b[34m/\u001b[0m\u001b[32m/\u001b[0m\u001b[31m/\u001b[0m \u001b]8;id=506359;https://multiqc.info\u001b\\\u001b[1mMultiQC\u001b[0m\u001b]8;;\u001b\\ üîç \u001b[2m| v1.12\u001b[0m\n",
      "\n",
      "\u001b[34m|           multiqc\u001b[0m | \u001b[33mMultiQC Version v1.16 now available!\u001b[0m\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/nikolay/BioGen/project/quality_check\n",
      "\u001b[34m|           multiqc\u001b[0m | Search path : /home/nikolay/BioGen/project/trimmed\n",
      "\u001b[2K\u001b[34m|\u001b[0m         \u001b[34msearching\u001b[0m | \u001b[38;2;114;156;31m‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ‚îÅ\u001b[0m \u001b[35m100%\u001b[0m \u001b[32m8/8\u001b[0m  [0m \u001b[32m3/8\u001b[0m \u001b[2mquality_check/SRR12905311.fastp.html\u001b[0m\n",
      "\u001b[?25h\u001b[34m|             fastp\u001b[0m | Found 1 reports\n",
      "\u001b[34m|            fastqc\u001b[0m | Found 2 reports\n",
      "\u001b[34m|           multiqc\u001b[0m | Compressing plot data\n",
      "\u001b[34m|           multiqc\u001b[0m | Report      : quality_check/multiqc_report.html\n",
      "\u001b[34m|           multiqc\u001b[0m | Data        : quality_check/multiqc_data\n",
      "\u001b[34m|           multiqc\u001b[0m | MultiQC complete\n",
      "+ firefox quality_check/multiqc_report.html\n",
      "+ mkdir -p index mappings\n",
      "+ bwa index ref/PAeruginosa.fasta\n",
      "[bwa_index] Pack FASTA... 0.03 sec\n",
      "[bwa_index] Construct BWT for the packed sequence...\n",
      "[bwa_index] 1.67 seconds elapse.\n",
      "[bwa_index] Update BWT... 0.02 sec\n",
      "[bwa_index] Pack forward-only FASTA... 0.02 sec\n",
      "[bwa_index] Construct SA from BWT and Occ... 0.62 sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa index ref/PAeruginosa.fasta\n",
      "[main] Real time: 2.464 sec; CPU: 2.367 sec\n",
      "++ nproc\n",
      "+ bwa mem ref/PAeruginosa.fasta trimmed/SRR12905311_1.fastq.gz trimmed/SRR12905311_2.fastq.gz -t 12\n",
      "[M::bwa_idx_load_from_disk] read 0 ALT contigs\n",
      "[M::process] read 1833476 sequences (120000028 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3012, 758289, 134, 2973)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 176, 237)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
      "[M::mem_pestat] mean and std.dev: (182.63, 81.66)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 275, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 555)\n",
      "[M::mem_pestat] mean and std.dev: (278.56, 84.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 666)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (147, 571, 2024)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5778)\n",
      "[M::mem_pestat] mean and std.dev: (1158.16, 1427.15)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7655)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 176, 235)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 459)\n",
      "[M::mem_pestat] mean and std.dev: (181.69, 80.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 571)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 1833476 reads in 64.386 CPU sec, 6.398 real sec\n",
      "[M::process] read 1833468 sequences (120000098 bp)...\n",
      "[M::process] read 1833442 sequences (120000111 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3214, 757316, 130, 3000)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 178, 235)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 459)\n",
      "[M::mem_pestat] mean and std.dev: (181.18, 79.61)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 571)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 275, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (2, 552)\n",
      "[M::mem_pestat] mean and std.dev: (278.18, 84.17)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 662)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (116, 948, 1790)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5138)\n",
      "[M::mem_pestat] mean and std.dev: (897.92, 1013.71)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6812)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (118, 173, 233)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 463)\n",
      "[M::mem_pestat] mean and std.dev: (179.43, 81.12)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 578)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1833468 reads in 67.764 CPU sec, 5.953 real sec\n",
      "[M::process] read 1833454 sequences (120000041 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3037, 757511, 140, 2944)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (121, 172, 230)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 448)\n",
      "[M::mem_pestat] mean and std.dev: (178.12, 78.59)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 557)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (221, 275, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 557)\n",
      "[M::mem_pestat] mean and std.dev: (278.36, 84.50)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 669)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (127, 450, 1385)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 3901)\n",
      "[M::mem_pestat] mean and std.dev: (581.15, 588.84)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5159)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (119, 176, 238)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 476)\n",
      "[M::mem_pestat] mean and std.dev: (181.58, 83.38)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 595)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1833442 reads in 67.688 CPU sec, 5.971 real sec\n",
      "[M::process] read 1833484 sequences (120000118 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3002, 756625, 134, 2918)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (122, 175, 234)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
      "[M::mem_pestat] mean and std.dev: (180.79, 80.96)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 570)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (221, 274, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (278.11, 84.46)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 665)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (132, 866, 1848)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5280)\n",
      "[M::mem_pestat] mean and std.dev: (935.12, 1172.67)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 6996)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (124, 177, 234)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 454)\n",
      "[M::mem_pestat] mean and std.dev: (180.49, 78.02)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 564)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1833454 reads in 68.613 CPU sec, 5.990 real sec\n",
      "[M::process] read 1833454 sequences (120000085 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3054, 758455, 122, 3041)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 176, 238)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 468)\n",
      "[M::mem_pestat] mean and std.dev: (183.03, 82.53)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (221, 274, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (277.76, 84.44)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 665)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (137, 408, 1458)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4100)\n",
      "[M::mem_pestat] mean and std.dev: (680.63, 692.34)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5421)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 177, 237)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 465)\n",
      "[M::mem_pestat] mean and std.dev: (181.92, 80.32)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 579)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1833484 reads in 67.396 CPU sec, 5.894 real sec\n",
      "[M::process] read 1356936 sequences (88811505 bp)...\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (3044, 756697, 175, 3034)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 177, 233)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 453)\n",
      "[M::mem_pestat] mean and std.dev: (179.98, 78.33)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 563)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (222, 275, 333)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 555)\n",
      "[M::mem_pestat] mean and std.dev: (278.41, 84.56)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 666)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (195, 1035, 1585)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 4365)\n",
      "[M::mem_pestat] mean and std.dev: (888.17, 815.92)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 5755)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 177, 236)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 458)\n",
      "[M::mem_pestat] mean and std.dev: (181.38, 79.54)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 569)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "[M::mem_process_seqs] Processed 1833454 reads in 67.426 CPU sec, 5.871 real sec\n",
      "[M::mem_pestat] # candidate unique pairs for (FF, FR, RF, RR): (2333, 560514, 109, 2327)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (125, 180, 238)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 464)\n",
      "[M::mem_pestat] mean and std.dev: (185.18, 81.99)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 577)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation FR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (221, 274, 332)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 554)\n",
      "[M::mem_pestat] mean and std.dev: (277.87, 84.29)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 665)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RF...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 979, 1884)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 5406)\n",
      "[M::mem_pestat] mean and std.dev: (1180.47, 1503.22)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 7193)\n",
      "[M::mem_pestat] analyzing insert size distribution for orientation RR...\n",
      "[M::mem_pestat] (25, 50, 75) percentile: (123, 177, 238)\n",
      "[M::mem_pestat] low and high boundaries for computing mean and std.dev: (1, 468)\n",
      "[M::mem_pestat] mean and std.dev: (181.25, 80.55)\n",
      "[M::mem_pestat] low and high boundaries for proper pairs: (1, 583)\n",
      "[M::mem_pestat] skip orientation FF\n",
      "[M::mem_pestat] skip orientation RF\n",
      "[M::mem_pestat] skip orientation RR\n",
      "[M::mem_process_seqs] Processed 1356936 reads in 49.722 CPU sec, 4.417 real sec\n",
      "[main] Version: 0.7.17-r1188\n",
      "[main] CMD: bwa mem -t 12 ref/PAeruginosa.fasta trimmed/SRR12905311_1.fastq.gz trimmed/SRR12905311_2.fastq.gz\n",
      "[main] Real time: 43.201 sec; CPU: 455.165 sec\n",
      "+ samtools view -b -o mappings/PAeruginosa.bam mappings/PAeruginosa.sam\n",
      "+ rm mappings/PAeruginosa.sam\n",
      "+ samtools flagstat mappings/PAeruginosa.bam\n",
      "12370620 + 0 in total (QC-passed reads + QC-failed reads)\n",
      "12357714 + 0 primary\n",
      "0 + 0 secondary\n",
      "12906 + 0 supplementary\n",
      "0 + 0 duplicates\n",
      "0 + 0 primary duplicates\n",
      "11069829 + 0 mapped (89.48% : N/A)\n",
      "11056923 + 0 primary mapped (89.47% : N/A)\n",
      "12357714 + 0 paired in sequencing\n",
      "6178857 + 0 read1\n",
      "6178857 + 0 read2\n",
      "10573080 + 0 properly paired (85.56% : N/A)\n",
      "10963636 + 0 with itself and mate mapped\n",
      "93287 + 0 singletons (0.75% : N/A)\n",
      "0 + 0 with mate mapped to a different chr\n",
      "0 + 0 with mate mapped to a different chr (mapQ>=5)\n",
      "+ samtools sort -@ 8 -o mappings/PAeruginosa.sorted.bam mappings/PAeruginosa.bam\n",
      "[bam_sort_core] merging from 0 files and 8 in-memory blocks...\n",
      "+ samtools index mappings/PAeruginosa.sorted.bam\n",
      "+ samtools faidx ref/PAeruginosa.fasta\n",
      "+ rm -rf bam_stats\n",
      "+ samtools stats mappings/PAeruginosa.sorted.bam\n",
      "+ plot-bamstats -p bam_stats/ mappings/PAeruginosa.sorted.bam.bc\n",
      "\"bam_stats/quals2.gp\" line 422: warning: y2 axis range undefined or overflow\n",
      "+ firefox bam_stats/index.html\n",
      "+ /home/nikolay/Apps/qualimap_v2.3/qualimap bamqc -bam mappings/PAeruginosa.sorted.bam\n",
      "Java memory size is set to 1200M\n",
      "Launching application...\n",
      "\n",
      "QualiMap v.2.3\n",
      "Built on 2023-05-19 16:57\n",
      "\n",
      "Selected tool: bamqc\n",
      "Available memory (Mb): 35\n",
      "Max memory (Mb): 1258\n",
      "Starting bam qc....\n",
      "Loading sam header...\n",
      "Loading locator...\n",
      "Loading reference...\n",
      "Number of windows: 400, effective number of windows: 400\n",
      "Chunk of reads size: 1000\n",
      "Number of threads: 12\n",
      "Processed 50 out of 400 windows...\n",
      "Processed 100 out of 400 windows...\n",
      "Processed 150 out of 400 windows...\n",
      "Processed 200 out of 400 windows...\n",
      "Processed 250 out of 400 windows...\n",
      "Processed 300 out of 400 windows...\n",
      "Processed 350 out of 400 windows...\n",
      "Processed 400 out of 400 windows...\n",
      "Total processed windows:400\n",
      "Number of reads: 12357714\n",
      "Number of valid reads: 11069829\n",
      "Number of correct strand reads:0\n",
      "\n",
      "Inside of regions...\n",
      "Num mapped reads: 11056923\n",
      "Num mapped first of pair: 5529437\n",
      "Num mapped second of pair: 5527486\n",
      "Num singletons: 93287\n",
      "Time taken to analyze reads: 18\n",
      "Computing descriptors...\n",
      "numberOfMappedBases: 718533766\n",
      "referenceSize: 6264404\n",
      "numberOfSequencedBases: 718448680\n",
      "numberOfAs: 121040005\n",
      "Computing per chromosome statistics...\n",
      "Computing histograms...\n",
      "Overall analysis time: 20\n",
      "end of bam qc\n",
      "Computing report...\n",
      "Writing HTML report...\n",
      "HTML report created successfully\n",
      "\n",
      "Finished\n",
      "+ firefox mappings/PAeruginosa.sorted_stats/qualimapReport.html\n",
      "+ mkdir -p variants\n",
      "+ freebayes -p 1 -f ref/PAeruginosa.fasta mappings/PAeruginosa.sorted.bam\n",
      "+ rm -rf vcf_stats\n",
      "+ bgzip -f variants/PAeruginosa.freebayes.vcf\n",
      "+ tabix -p vcf variants/PAeruginosa.freebayes.vcf.gz\n",
      "+ bcftools stats -F ref/PAeruginosa.fasta -s - variants/PAeruginosa.freebayes.vcf.gz\n",
      "+ plot-vcfstats -p vcf_stats/ variants/PAeruginosa.freebayes.vcf.gz.stats\n",
      "Parsing bcftools stats output: variants/PAeruginosa.freebayes.vcf.gz.stats\n",
      "Plotting graphs: python3 plot.py\n",
      "Creating PDF: pdflatex summary.tex >plot-vcfstats.log 2>&1\n",
      "Finished: vcf_stats//summary.pdf\n",
      "+ firefox vcf_stats/summary.pdf\n",
      "+ igv -g ref/PAeruginosa.fasta mappings/PAeruginosa.sorted.bam variants/PAeruginosa.freebayes.vcf.gz GSE160188_core.vcf.gz ref/PAeruginosa.gff\n",
      "INFO [Oct 10,2023 18:25] [Main] Startup  IGV Version user not_set\n",
      "INFO [Oct 10,2023 18:25] [Main] Java 17.0.8.1 (build 17.0.8.1+1-Ubuntu-0ubuntu122.04) 2023-08-24\n",
      "INFO [Oct 10,2023 18:25] [Main] Java Vendor: Private Build Unknown\n",
      "INFO [Oct 10,2023 18:25] [Main] JVM: OpenJDK 64-Bit Server VM    \n",
      "INFO [Oct 10,2023 18:25] [Main] OS: Linux 5.15.0-86-generic amd64\n",
      "INFO [Oct 10,2023 18:25] [Main] IGV Directory: /home/nikolay/igv\n",
      "WARNING [Oct 10,2023 18:25] [GenomeLoader] Sequence file not found: /home/nikolay/igv/genomes/seq/Y55\n",
      "INFO [Oct 10,2023 18:25] [GenomeManager] Loading genome: ref/PAeruginosa.fasta\n",
      "INFO [Oct 10,2023 18:25] [TrackLoader] Loading resource:  mappings/PAeruginosa.sorted.bam\n",
      "INFO [Oct 10,2023 18:25] [TrackLoader] Loading resource:  variants/PAeruginosa.freebayes.vcf.gz\n",
      "INFO [Oct 10,2023 18:25] [TrackLoader] Loading resource:  ref/PAeruginosa.gff\n",
      "INFO [Oct 10,2023 18:25] [ShutdownThread] Shutting down\n",
      "+ mv mappings/PAeruginosa.sorted_stats ./qualimap_report\n",
      "mv: cannot move 'mappings/PAeruginosa.sorted_stats' to './qualimap_report/PAeruginosa.sorted_stats': Directory not empty\n",
      "+ '[' -d /mnt/c/Users/Asus/BioGen ']'\n",
      "+ mkdir -p data\n",
      "+ mv ref/PAeruginosa.fasta mappings/PAeruginosa.sorted.bam mappings/PAeruginosa.sorted.bam.bai mappings/PAeruginosa.sorted.bam.bc variants/PAeruginosa.freebayes.vcf.gz GSE160188_core.vcf.gz ref/PAeruginosa.gff data\n",
      "mv: cannot stat 'GSE160188_core.vcf.gz': No such file or directory\n",
      "+ rm -rf variants mappings index assembly trimmed ref\n"
     ]
    }
   ],
   "source": [
    "!./project.sh"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "1b359156",
   "metadata": {},
   "source": [
    "<h1>–†–µ–∑—É–ª—å—Ç–∞—Ç—ã:</h1>\n",
    "<h2>–ê–Ω–Ω–æ—Ç–∞—Ü–∏—è:</h2>\n",
    "<ul>\n",
    "    <li> <a href=\"https://www.ncbi.nlm.nih.gov/sra/SRR12905311\" target=\"_blank\"> SRX9370194: GSM4862160: PAO1_MIC_Exp5_t=72 (DNA-seq); Pseudomonas aeruginosa; OTHER </a> </li>\n",
    "    <li> <b>–ò—Å—Å–ª–µ–¥–æ–≤–∞–Ω–∏–µ:</b> –ü–æ–≤—ã—à–∞–µ—Ç –ª–∏ –º–µ–¥—å —É—Å—Ç–æ–π—á–∏–≤–æ—Å—Ç—å –∫ –∞–Ω—Ç–∏–±–∏–æ—Ç–∏–∫–∞–º —É —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω—ã—Ö –±–∞–∫—Ç–µ—Ä–∏–π –∏/–∏–ª–∏ –∏–Ω–¥—É—Ü–∏—Ä—É–µ—Ç –∞–Ω—Ç–∏–±–∏–æ—Ç–∏–∫–æ—Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å —É —á—É–≤—Å—Ç–≤–∏—Ç–µ–ª—å–Ω—ã—Ö –±–∞–∫—Ç–µ—Ä–∏–π? </li>\n",
    "    <li> <b>–û—Ä–≥–∞–Ω–∏–∑–º:</b> <a href=\"https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?mode=Info&id=287\" target=\"_blank\"> <i> Pseudomonas aeruginosa (–°–∏–Ω–µ–≥–Ω–æ–π–Ω–∞—è –ø–∞–ª–æ—á–∫–∞) </i> </a> </li>\n",
    "    <li> –®—Ç–∞–º–º: <a href=\"https://www.ncbi.nlm.nih.gov/Taxonomy/Browser/wwwtax.cgi?id=208964&mode=info\" target=\"_blank\"> PAO1 (–¥–∏–∫–∏–π —Ç–∏–ø) </a> </li>\n",
    "    <li> –†–µ—Ñ–µ—Ä–µ–Ω—Å–Ω—ã–π –≥–µ–Ω–æ–º: <a href=\"https://www.ncbi.nlm.nih.gov/datasets/genome/GCF_000006765.1\" target=\"_blank\"> ASM676v1 </a> </li>\n",
    "    <li> GEO Accession: <a href=\"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSM4862160\" target=\"_blank\"> GSM4862160 </a> </li>\n",
    "    <li> SRA: <a href=\"https://trace.ncbi.nlm.nih.gov/Traces/?view=study&acc=SRP288731\" target=\"_blank\"> SRP288731 </a> </li>\n",
    "    <li> BioProject: <a href=\"https://www.ncbi.nlm.nih.gov/bioproject/PRJNA672287\" target=\"_blank\"> PRJNA672287 </a> </li>\n",
    "    <li> GEO: <a href=\"https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE160188\" target=\"_blank\"> GSE160188 </a> </li>\n",
    "</ul>\n",
    "<h2>–û—Ç—á—ë—Ç—ã:</h2>\n",
    "<p>\n",
    "    <a href=\"quality_check/SRR12905311.fastp.html\" target=\"_blank\"> –†–µ–∑—É–ª—å—Ç–∞—Ç—ã —Ç—Ä–∏–º–º–∏–Ω–≥–∞ </a> <br/>\n",
    "    <a href=\"quality_check/SRR12905311_1_fastqc.html\" target=\"_blank\"> –û—Ü–µ–Ω–∫–∞ –∫–∞—á–µ—Å—Ç–≤–∞ (—Ä–∏–¥ 1) </a> <br/>\n",
    "    <a href=\"quality_check/SRR12905311_2_fastqc.html\" target=\"_blank\"> –û—Ü–µ–Ω–∫–∞ –∫–∞—á–µ—Å—Ç–≤–∞ (—Ä–∏–¥ 2) </a> <br/>\n",
    "    <a href=\"quality_check/multiqc_report.html\" target=\"_blank\">  –ü—Ä–æ–≤–µ—Ä–∫–∞ –∫–∞—á–µ—Å—Ç–≤–∞ (—Å–≤–æ–¥–Ω—ã–π –æ—Ç—á—ë—Ç) </a> <br/>\n",
    "    <a href=\"assembly_report.pdf\" target=\"_blank\"> –°–±–æ—Ä–∫–∞ —Ä–∏–¥–æ–≤ </a> <br/>\n",
    "    <a href=\"bam_stats/index.html\" target=\"_blank\"> –í—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏–µ —Ä–∏–¥–æ–≤ –Ω–∞ —Ä–µ—Ñ–µ—Ä–µ–Ω—Å </a> <br/>\n",
    "    <a href=\"qualimap_report/qualimapReport.html\" target=\"_blank\"> –ü—Ä–æ–≤–µ—Ä–∫–∞ –∫–∞—á–µ—Å—Ç–≤–∞ –≤—ã—Ä–∞–≤–Ω–∏–≤–∞–Ω–∏—è </a> <br/>\n",
    "    <a href=\"vcf_stats/summary.pdf\" target=\"_blank\"> –ü–æ–∏—Å–∫ SNP </a> <br/>\n",
    "    <a href=\"https://igv.org\" target=\"_blank\">IGV</a>: <code> igv -g data/PAeruginosa.fasta data/PAeruginosa.sorted.bam data/PAeruginosa.freebayes.vcf.gz data/PAeruginosa.gff </code> <br/>\n",
    "    <a href=\"https://github.com/tseemann/abricate\" target=\"_blank\">ABRicate</a>: <a href=\"abricate_report.tsv\" target=\"_blank\"> –ù–∞–ª–∏—á–∏–µ –≥–µ–Ω–æ–≤ —É—Å—Ç–æ–π—á–∏–≤–æ—Å—Ç–∏ –∫ –ø—Ä–æ—Ç–∏–≤–æ–º–∏–∫—Ä–æ–±–Ω—ã–º –ø—Ä–µ–ø–∞—Ä–∞—Ç–∞–º </a>\n",
    "    <p><i> –ö–æ–º–º–µ–Ω—Ç–∞—Ä–∏–π:\n",
    "        –î–ª—è —É–ª—É—á—à–µ–Ω–∏—è –∫–∞—á–µ—Å—Ç–≤–∞ —Ä–∏–¥–æ–≤ –±—ã–ª–∞ –ø—Ä–æ–≤–µ–¥–µ–Ω–∞ –æ–ø–µ—Ä–∞—Ü–∏—è —Ç—Ä–∏–º–º–∏–Ω–≥–∞.  \n",
    "        –í —Ä–µ–∑—É–ª—å—Ç–∞—Ç–µ —Ä–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏–µ ATGC –≤ –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç—è—Ö (<a href=\"quality_check/SRR12905311_1_fastqc.html#M4\">Per base sequence content</a>) —Å—Ç–∞–ª–æ –±–æ–ª–µ–µ —Ä–∞–≤–Ω–æ–º–µ—Ä–Ω—ã–º.\n",
    "        –†–∞—Å–ø—Ä–µ–¥–µ–ª–µ–Ω–∏–µ –¥–ª–∏–Ω—ã –ø–æ—Å–ª–µ–¥–æ–≤–∞—Ç–µ–ª—å–Ω–æ—Å—Ç–∏ (<a href=\"quality_check/SRR12905311_1_fastqc.html#M7\">Sequence Length Distribution</a>)\n",
    "        –∏ —Å–æ–¥–µ—Ä–∂–∞–Ω–∏–µ GC (<a href=\"quality_check/SRR12905311_1_fastqc.html#M5\">Per sequence GC content</a>) –≤–ø–æ–ª–Ω–µ —Å–Ω–æ—Å–Ω–æ–µ –¥–ª—è –¥–∞–ª—å–Ω–µ–π—à–µ–≥–æ –∞–Ω–∞–ª–∏–∑–∞.\n",
    "    </i></p>\n",
    "</p>\n",
    "<h2>–ì–µ–Ω—ã –∏ –º—É—Ç–∞—Ü–∏–∏:</h2>\n",
    "<table>\n",
    "    <tr>\n",
    "        <th>–ì–µ–Ω</th>\n",
    "        <th>Symbol</th>\n",
    "        <th>Locus Tag</th>\n",
    "        <th>IGV window</th>\n",
    "        <th>–ü—Ä–∏–º–µ—Ä –º—É—Ç–∞—Ü–∏–∏</th>\n",
    "        <th>–ö–æ–º–º–µ–Ω—Ç–∞—Ä–∏–π</th>\n",
    "    </tr>\n",
    "    <tr>\n",
    "        <td><a href=\"https://www.ncbi.nlm.nih.gov/gene/880226\">aminoglycoside 3-N-acetyltransferase</a></td>\n",
    "        <td>aph</td>\n",
    "        <td>PA4119</td>\n",
    "        <td><img src=\"pics/PA4119.png\"></td>\n",
    "        <td>Type: MNP<br/>Position: 4607781-4607788<br/>Reference: GCAGATTT*<br/>Alternate: AGAGGTTG</td>\n",
    "        <td>–ü—Ä–µ–¥–ø–æ–ª–æ–∂–∏—Ç–µ–ª—å–Ω–æ –≥–µ–Ω –≤–ª–∏—è–µ—Ç –Ω–∞ —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å –∫ <a href=\"https://ru.wikipedia.org/wiki/%D0%9A%D0%B0%D0%BD%D0%B0%D0%BC%D0%B8%D1%86%D0%B8%D0%BD\">KANAMYCIN</a>. –ú–Ω–æ–∂–µ—Å—Ç–≤–µ–Ω–Ω—ã–µ –º—É—Ç–∞—Ü–∏–∏ –≤ –≥–µ–Ω–µ –≤–µ—Ä–æ—è—Ç–Ω–æ –ø–æ–≤–ª–∏—è–ª–∏ –Ω–∞ —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å.</td>\n",
    "    </tr>\n",
    "    <tr>\n",
    "        <td><a href=\"https://www.ncbi.nlm.nih.gov/gene/878374\">chloramphenicol acetyltransferase</a></td>\n",
    "        <td>cat</td>\n",
    "        <td>PA0706</td>\n",
    "        <td><img src=\"pics/PA0706.png\"></td>\n",
    "        <td>Type: MNP<br/>Position: 779812-779814<br/>Reference: ACA*<br/>Alternate: GCC</td>\n",
    "        <td>–ü—Ä–µ–¥–ø–æ–ª–æ–∂–∏—Ç–µ–ª—å–Ω–æ –≥–µ–Ω –≤–ª–∏—è–µ—Ç –Ω–∞ —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å –∫ <a href=\"https://ru.wikipedia.org/wiki/%D0%A5%D0%BB%D0%BE%D1%80%D0%B0%D0%BC%D1%84%D0%B5%D0%BD%D0%B8%D0%BA%D0%BE%D0%BB\">CHLORAMPHENICOL</a>. –ù–µ—Å–∫–æ–ª—å–∫–æ –º—É—Ç–∞—Ü–∏–π –≤ –≥–µ–Ω–µ –º–æ–≥–ª–∏ –ø–æ–≤–ª–∏—è—Ç—å –Ω–∞ —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å.</td>\n",
    "    </tr>\n",
    "    <tr>\n",
    "        <td><a href=\"https://www.ncbi.nlm.nih.gov/gene/877785\">glutathione transferase FosA</a></td>\n",
    "        <td>-</td>\n",
    "        <td>PA1129</td>\n",
    "        <td><img src=\"pics/PA1129.png\"></td>\n",
    "        <td>Type: SNP<br/>Position: 1222092<br/>Reference: –°*<br/>Alternate: T</td>\n",
    "        <td>–ü—Ä–µ–¥–ø–æ–ª–æ–∂–∏—Ç–µ–ª—å–Ω–æ –≥–µ–Ω –≤–ª–∏—è–µ—Ç –Ω–∞ —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å –∫ <a href=\"https://ru.wikipedia.org/wiki/%D0%A4%D0%BE%D1%81%D1%84%D0%BE%D0%BC%D0%B8%D1%86%D0%B8%D0%BD\">FOSFOMYCIN</a>. –ú–∞–ª–æ–≤–µ—Ä–æ—è—Ç–Ω–æ, —á—Ç–æ –æ–¥–Ω–∞ –º—É—Ç–∞—Ü–∏—è —Å–µ—Ä—å—ë–∑–Ω–æ –ø–æ–≤–ª–∏—è–ª–∞ –Ω–∞ —Ä–µ–∑–∏—Å—Ç–µ–Ω—Ç–Ω–æ—Å—Ç—å.</td>\n",
    "    </tr>\n",
    "</table>"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "6513efe4",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
